src/sync/incomingSyncUtils.js

```diff
@@ -306,9 +306,32 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
   let internalRecord;
   switch (recordType) {
     case 'IndicatorAttribute': {
+      internalRecord = {
+        id: record.ID,
+        indicator: database.getOrCreate('ProgramIndicator', record.indicator_ID),
+        description: record.description,
+        code: record.code,
+        index: parseNumber(record.index),
+        isRequired: parseBoolean(record.is_required),
+        valueType: record.data_type && record.data_type.value,
+        valueDefault: record.data_type && record.data_type.default,
+        axis: record.axis,
+        isActive: parseBoolean(record.is_active),
+      };
+      database.update(recordType, internalRecord);
       break;
     }
     case 'IndicatorValue': {
+      const recordValue = record.value && JSON.parse(record.value);
```

Comment:
`value` is required and this has the possibility of assigning a null/undefined. I think `(record.value && JSON.parse(record.value)) || "" ` might be a bit safer

Reply:
Good catch, e7f9bbccdc2a2ac8fc471a33acaa4f58565c7a34.